### PR TITLE
PPNA-6416 Real-time query API

### PR DESCRIPTION
**Pull request by mrfijas**
Copy/paste of query API with `date_from/date_to -> lookback_window` change

### CODE REVIEW

File: custom_reports/http_api/definitions/real_time_query_request.yaml [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+            Optional ID of the transformation. Must be supported by the
+            column that was specified.
+        goal_id:
+          type: integer
+          description: |
+            Optional ID of a goal. Available only for metrics related to goals.
+      required:
+        - column_id
+    example: [{"column_id":"device_type"},{"column_id":"revenue","transformation_id":"sum"}]
+  lookback_window:
```
Comment (pulina) [inline custom_reports/http_api/definitions/real_time_query_request.yaml:29]: Now I have few seconds to wander why we have `lookback_window` here? We have it case we implement it this way here https://github.com/PiwikPRO/PPMS-Analytics-Reporting/commit/f12eb73fe3512bc0563cea8af599ccb170e289a4. But on grooming we [decided to go with date_time approach](https://piwikpro.atlassian.net/wiki/spaces/PPNA/pages/86346957059/2024-01-30+Public+API+in+RTR). Is this bug  that slipped in or we have some further discussion and make different decisions.

**Review by pulina — APPROVED** [review]

File: custom_reports/http_api/definitions/real_time_query_request.yaml (hunk 1 above)
Comment (mrfijas) [reply custom_reports/http_api/definitions/real_time_query_request.yaml:29]: raw data API uses date_from/date_to, query API uses lookback_window

File: custom_reports/http_api/definitions/real_time_query_request.yaml [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+    type: int
+    format: date
```
Comment (janusz-piechota-piwik) [inline custom_reports/http_api/definitions/real_time_query_request.yaml:31]: Is it correct ? Format date is for string (for example, 2017-07-21) https://swagger.io/docs/specification/data-models/data-types/#format

Comment (mrfijas) [reply custom_reports/http_api/definitions/real_time_query_request.yaml:31]: of course not, good catch :+1:

**Review by janusz-piechota-piwik — APPROVED** [review]

**Pull request merged**